File: app/src/main/java/com/example/marcin/meetfriends/ui/chat/ChatModule.java
package com.example.marcin.meetfriends.ui.chat;

import com.example.marcin.meetfriends.ui.common.params.EventBasicInfoParams;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

/**
 * Created by marci on 2017-12-03.
 */
@Module
abstract public class ChatModule {

  @Binds
  abstract ChatContract.View bindView(ChatActivity view);

  @Binds
  abstract ChatContract.Presenter bindPresenter(ChatPresenter presenter);

  @Provides
  static EventBasicInfoParams provideBasicInfoParams(ChatActivity view) {
    return new EventBasicInfoParams(view.getIntent().getExtras());
  }
}

File: app/src/main/java/com/example/marcin/meetfriends/ui/chat/ChatModule.kt
package com.example.marcin.meetfriends.ui.chat

import dagger.Binds
import dagger.Module

/**
 * Created by marci on 2017-11-21.
 */
@Module
abstract class ChatModule {

  @Binds
  abstract fun bindView(view: ChatActivity): ChatContract.View

  @Binds
  abstract fun bindPresenter(presenter: ChatPresenter): ChatContract.Presenter
}